game_event/models.py

```diff
@@ -17,3 +21,29 @@ class GameEvent(models.Model):
         validators=[MinLengthValidator(4)],
         choices=BallGame.choices
     )
+
+    @staticmethod
+    def create(time, level_of_game, min_number_of_players, max_number_of_players, court, ball_game):
+        GameEvent.validate_game_event(court, ball_game, min_number_of_players, max_number_of_players, time)
+        game_event = GameEvent(time=time,
+                               level_of_game=level_of_game,
+                               min_number_of_players=min_number_of_players,
+                               max_number_of_players=max_number_of_players,
+                               court=court,
+                               ball_game=ball_game)
+        game_event.save()
+        return game_event
+
+    @staticmethod
+    def validate_game_event(court, ball_game, min_number_of_players, max_number_of_players, time):
+        errors = []
+        if not CourtBallGame.is_ball_game_playable(court, ball_game):
+            errors.append("The selected court doesn't support the selected ball game.")
+        if min_number_of_players < 2:
```

Comment:
Can't this validation be part of the model?

Reply:
not sure i understood
i wrap the call with the GameEvent.is_ball_game_playable_at_court()
if it wasnt your intension then i would need elaboration on this part